### ISSUE TITLE
Random Spotify Sync Failed Error

### ISSUE THREAD

**Issue by OzGav** [music-assistant/hass-music-assistant#742]
### What version of Music Assistant has the issue?

2022.7.1

### The problem

Every so often (infrequently) I see the error as shown in the log below. Unfortunately at this time I don't have much more info to provide. Submitted for info.

### How to reproduce

Not sure at this time.

### Relevant log output

Logger: music_assistant
Source: /usr/local/lib/python3.10/site-packages/music_assistant/mass.py:247
First occurred: 2:04:17 PM (1 occurrences)
Last logged: 2:04:17 PM

Job [Library sync for provider Spotify] failed with error .
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/music_assistant/music_providers/spotify.py", line 551, in _get_token
    await librespot.wait()
  File "/usr/local/lib/python3.10/asyncio/subprocess.py", line 134, in wait
    return await self._transport._wait()
  File "/usr/local/lib/python3.10/asyncio/base_subprocess.py", line 235, in _wait
    return await waiter
asyncio.exceptions.CancelledError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.10/asyncio/tasks.py", line 456, in wait_for
    return fut.result()
asyncio.exceptions.CancelledError

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/music_assistant/models/music_provider.py", line 398, in sync_library
    async for prov_item in self._get_library_gen(media_type)():
  File "/usr/local/lib/python3.10/site-packages/music_assistant/music_providers/spotify.py", line 150, in get_library_artists
    spotify_artists = await self._get_data(
  File "/usr/local/lib/python3.10/site-packages/music_assistant/music_providers/spotify.py", line 627, in _get_data
    token = await self.get_token()
  File "/usr/local/lib/python3.10/site-packages/music_assistant/music_providers/spotify.py", line 514, in get_token
    tokeninfo = await asyncio.wait_for(self._get_token(), 5)
  File "/usr/local/lib/python3.10/asyncio/tasks.py", line 458, in wait_for
    raise exceptions.TimeoutError() from exc
asyncio.exceptions.TimeoutError

### What version of Home Assistant Core are your running

2022.7.6

### What type of installation are you running?

Home Assistant OS

### On what type of hardware are you running?

Raspberry Pi

**Comment by marcelveldt** [comment]
Do you have a slow internet connection perhaps or just some random instabilities sometimes ?

**Comment by OzGav** [comment]
Not at your European standards but I get 50M and yes it could be a random event. Just thought it would be worth mentioning in case the log meant more to you than I.

**Comment by OzGav** [comment]
Maybe it is occuring on each sync. I see another entry in the log. I am assuming it is a routine sync as I was't home at the time:
```
First occurred: 2:04:17 PM (2 occurrences)
Last logged: 8:19:10 PM
```

**Comment by erkr** [comment]
Yes, MA periodically runs a sync. I recall every 3 hours, so one sync, in between was successful then.

**Comment by OzGav** [comment]
Yes quite right. It isn't random but it is inconsistent. I have now seen three in eighteen hours
```
First occurred: July 26, 2022 at 2:04:17 PM (3 occurrences)
Last logged: 5:19:10 AM
```
As this is coming good at some point I think it would be good if this could be caught, ignored and logged for info with a comment that it will be retried in three hours?

**Comment by marcelveldt** [comment]
The error is thrown if 5 retries failed. Looks like in some cases Spotify can be a bit slow to respond so I've extended the timeout and retries a bit. Fix available in 2022.7.2
